staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go

```diff
@@ -449,14 +448,10 @@ func (nm *NodeManager) GetNodeInfoWithNodeObject(node *v1.Node) (NodeInfo, error
 }
 
 func (nm *NodeManager) CredentialManager() *SecretCredentialManager {
-	nm.credentialManagerLock.Lock()
-	defer nm.credentialManagerLock.Unlock()
 	return nm.credentialManager
 }
 
 func (nm *NodeManager) UpdateCredentialManager(credentialManager *SecretCredentialManager) {
-	nm.credentialManagerLock.Lock()
-	defer nm.credentialManagerLock.Unlock()
 	nm.credentialManager = credentialManager
```

Comment:
golang does not guarantee direct pointer assignments (without constructs) to be atomic and by the looks of things NodeManager does need concurrency. could you please explain why do you think the locks are not needed here?

Reply:
@neolit123 , thanks for the review. I tested both the scenarios with go's -race compilation flag. The assignment has a race condition, while the single return statement doesn't haven't. I created two examples to validate the hypothesis of the atomic operations for two scenarios:

1. assignment to a pointer in struct: https://play.golang.com/p/q4t3TD5BQ8d
2. returning a pointer from a struct: https://play.golang.com/p/T4v_C7D39ZU

if I build with: go build -race <program>, then the first one detects the race condition while the return doesn't detect race condition.

I think we can remove lock on the return but not on assignment. Let me know your thoughts.

Reply:
yet, it's not thread safe to to read the pointer while another thread is writing to it.

one potential improvement here may be [rwmutex](https://golang.org/pkg/sync/#RWMutex), but i would defer to the owners on that.

Reply:
@neolit123 , thanks. I agree that it would need atleast an rlock() if its not an atomic (just in case it gets deleted in the middle of fetching). In my return-pointer-value example with -race option, I am not deleting any pointer.